### PR TITLE
Add "warnings" from prometheus upstream

### PR DESCRIPTION
**Pull request by jacksontj**
Split from #141 
TLDR; upstream prometheus added a "storage.Warnings" which it uses for non-primary datasource reads. This pr is for adding support for such a thing

References/links:
- I~~nitial add of warnings to client (https://github.com/prometheus/client_golang/pull/562 )~~
- ~~rework of warnings in api client (https://github.com/prometheus/client_golang/pull/599)~~
- add warnings to series (https://github.com/prometheus/client_golang/pull/603)
- labelsName/labelValues warnings? (https://github.com/prometheus/prometheus/issues/5661)
- better configuration for warnings (https://github.com/prometheus/prometheus/issues/5662)

### CODE REVIEW

**Comment by jacksontj** [comment]
Still working through these warnings, at this point I have PRs open for all the ones that are in prom (although it seems that some *should* have this warning thing...).

Unfortuantely this is a large code change to add warnings as it changes the cleint interface (which means changes at all the layers). 

In addition this means we need to determine when *promxy* will return warnings and not errors :/
Until this point I've been treating any error as fatal, so I'm thinking the "ignore_error" will get a similar companion of "warning_error" (or something similar) that takes al errors and marks them as warnings. This way the client (in config) can be explicit about what failures are tolerable-- and I'll continue to defaulting to requiring non-errors.


Lastly (for now) it also means I need to rework some of the merging logic to take warnings into consideration, if one node returned a "warning" and has no holes, we don't want it to be the primary assuming another one has no holes and has no warnings. Because of these large changes I'm thinking I'll actually split out the
